OdyseeTeam/odysee-frontend
# Improve local tab syncing

## Status
:yellow_circle: Needs more testing @ `kp`

## Issue
- Closes #2037

## Overview
- Changes to synced settings like theme and private playlists are now immediate between local tabs.  This reduces any sync error from the existing API-based mechanism, which doesn't kick in until 2s after the tab is focused, potentially causing merge conflicts.
- The current API-based syncing remain as-is. The idea is that if the changes are done across all tabs, the API-calls will end up being a no-op
    - Edit: will also try to see if we can reduce the calls

## Test
- All settings that are currently saved to the wallet should be tab-synced.
- Additions so far:  
    - Deleting a notification
    - Player seekbar position (already there previously, but re-implemented the new way).
    - _...can add more if you guys find any useful ones_
- Ensure per-tab settings like navigation paths and form values (e.g. comments) are not being overwritten by changes in another tab.
    - Test out as many settings as possible.
        - Publishing? SDK calls?
- Ensure the existing wallet sync between devices isn't broken.
- Note that the following will remain per-tab:
    - Queue (just like other sites)
